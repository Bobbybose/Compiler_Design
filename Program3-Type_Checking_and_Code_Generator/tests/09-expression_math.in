int a = 2;
float b = 3.1;
char c = 'a';
int d;
int e;
d = a + b;
e = d - c;
e = d * a;
e = d / a;
