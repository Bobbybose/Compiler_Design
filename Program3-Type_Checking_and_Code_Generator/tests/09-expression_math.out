Code:
%t1 = 2
a = %t1
%t2 = 3.100000
b = %t2
%t3 = 'a'
c = %t3
%t4 = int2float a
%t5 = %t4 + b
%t6 = float2int %t5
d = %t6
%t7 = char2int c
%t8 = d - %t7
%t9 = %t8
e = %t9
%t10 = d * a
%t11 = %t10
e = %t11
%t12 = d / a
%t13 = %t12
e = %t13

